### PR TITLE
Fibonacci Search in C++

### PR DESCRIPTION
**Pull request by maneprajakta**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

I have added Fibonacci search under the search section in c++.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2242

### CODE REVIEW

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+2.
+Enter Size Of Arrary:
+4
+Enter Elements In Array
+1 2 3 4 
+Enter The Element To Be Searched
+5
+Element Not Found
+
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Remove this. One sample I/O does the trick

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    cin>>element;
+
+    int index = fibonacciSearch(arr,n,element); /* Call fibonacci search function */
+
+    if(index==-1)
+    cout<<"Element Not Found"<<"\n";   /* if element is not present */
+    else
+    cout<<"Element Found At Array Index : "<<index;   /* if Present print index*/
+
+   return 0;
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Indent the Code. In fact, run a formatter on the whole code

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    else if(arr[i]>element)
+    {
+        fibM = fibM2;
+        fibM1 = fibM1 - fibM2;
+        fibM2 = fibM - fibM1;
+    }
+    /* if found return index */
+    else return i;
+    } 
+    /* check for last array element */
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Use `// .......` for single line comments only

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      we move 3 fibonacci variable ,2 fibonacci down
+      we drop approx rear 2/3 of remaining array 
+    */
+    else if(arr[i]>element)
+    {
+        fibM = fibM2;
+        fibM1 = fibM1 - fibM2;
+        fibM2 = fibM - fibM1;
+    }
+    /* if found return index */
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Use `// .......` for single line comments only

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+F(2) = F(1) + F(0) = 0+1 = 1 
+F(n) = F(n-1)+F(n-2) for n>=2
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+int fibonacciSearch(int arr[],int n,int element)
+{   
+    /* Initializing fibonacci no */
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Use `// .......` for single line comments only

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 6]
```diff
@@ -0,0 +1,132 @@
+/*
+                                              Fibonacci Search
+
+Search for an element in a Sorted Array
+
+Time Complexity : O(log(n))
+Auxiliary Space : O(1)
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Keep complexities at the end of the code

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Code looks good. Stylistic changes requested 👍

File: C-Plus-Plus/search/fibonacci_search.cpp (hunk 1 above)
Comment (maneprajakta) [reply C-Plus-Plus/search/fibonacci_search.cpp]: okay

**Comment by maneprajakta** [comment]
Sure!
Will do all the Changes and Come back to you again.

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+1 6 10 11 78   
+Enter The Element To Be Searched
+78
+Element Found At Array Index :
+
+Algorithm Analysis:
+Time Complexity : O(log(n))
+Auxiliary Space : O(1) 
+
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Add a new line at the EOF

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+   int element; // Element that user want's to search 
+   cout << "Enter The Element To Be Searched" << "\n";
+   cin >> element;
+
+   int index = fibonacciSearch(arr, n, element); // Call fibonacci search function 
+
+   if (index == -1)
+      cout << "Element Not Found" << "\n"; // if element is not present 
+   else
+      cout << "Element Found At Array Index : " << index; // if Present print index
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Use comments above the line of code not beside it.

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+   }
+
+   int element; // Element that user want's to search 
+   cout << "Enter The Element To Be Searched" << "\n";
+   cin >> element;
+
+   int index = fibonacciSearch(arr, n, element); // Call fibonacci search function 
+
+   if (index == -1)
+      cout << "Element Not Found" << "\n"; // if element is not present
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Use comments above the line of code not beside it.

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+   for (int i = 0; i < n; i++) {
+      cin >> arr[i];
+   }
+
+   int element; // Element that user want's to search 
+   cout << "Enter The Element To Be Searched" << "\n";
+   cin >> element;
+
+   int index = fibonacciSearch(arr, n, element); // Call fibonacci search function
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Use comments above the line of code not beside it.

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+   cin >> n;
+
+   int arr[n]; // array declaration 
+   cout << "Enter Elements In Array" << "\n";
+
+   for (int i = 0; i < n; i++) {
+      cin >> arr[i];
+   }
+
+   int element; // Element that user want's to search
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Use comments above the line of code not beside it.

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+   // check for last array element 
+   if (fibM1 && arr[offset + 1] == element)
+      return offset + 1;
+
+   // element not found return -1
+   return -1;
+}
+
+int main() {
+   int n; // size of array
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Use comments above the line of code not beside it.

File: C-Plus-Plus/search/fibonacci_search.cpp [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+   // element not found return -1
+   return -1;
+}
+
+int main() {
+   int n; // size of array 
+   cout << "Enter Size Of Arrary:" << "\n";
+   cin >> n;
+
+   int arr[n]; // array declaration
```
Comment (HarshCasper) [inline C-Plus-Plus/search/fibonacci_search.cpp]: Use comments above the line of code not beside it.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Don't use redundant comments. We don't need to explain what each variable or console statement do

**Review by HarshCasper — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]
Looks good to me. Thank you for your contribution. 👍

**Pull request merged**